### PR TITLE
Fix Rename-Item to allow Unix globbing patterns inpaths (#2799)

### LINKED ISSUE #2799: *-Item should handle files with wildcard characters
Steps to reproduce
------------------
```powershell
Describe "Rename-Item" {
    It "Rename-Item should work with files which have special characters" {
        $file = new-item -itemtype file "${TESTDRIVE}/file[txt].txt"
        test-path -literal $file.fullname | Should be $true
        Rename-Item -literal $file.fullname "${TESTDRIVE}/file.txt.txt"
        test-path -literal $file.fullname | Should be $false
        test-path -literal "${TESTDRIVE}/file.txt.txt" | should be $true
    }
}
```

Expected behavior
-----------------
this test should pass (and it does on Windows)

Actual behavior
---------------
Rename-Item can't seem to find the file
```
Describing Rename-Item
 [-] Rename-Item should work with files which have special characters 1.41s
   An object at the specified path /tmp/bc945451-748f-471e-8dab-f2d09a1b2221/file[txt].txt does not exist.
   at line: 5 in /home/james/tmp/rename.item.tests.ps1
Tests completed in 1.41s
Passed: 0 Failed: 1 Skipped: 0 Pending: 0
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.13-10-g7015e66baecc02955640b5c19e7feb5cb6746c63
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
```
uname -a
Linux jimtru-ub16d 4.4.0-47-generic #68-Ubuntu SMP Wed Oct 26 19:39:52 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

### PR DESCRIPTION
**Pull request by jeffbi**
Fixes #2799

In the process of normalizing a relative path, PowerShell checks to see
see if the path exists, which it does by invoking `Directory.EnumerateFiles(directory, filename);`

On Unix platforms, if the filename contains globbing patterns, such as `[ab]`,
`EnumerateFiles` (and `EnumerateDirectories`) will perform the globbing. Using
globbing patterns, a file named `file[txt].txt` is reported as not existing.

This fix changes the file-existence test on Unix to use a native function
instead of either of the Directory.EnumerateXXX functions.

This affects the `Move-Item`, `Remove-Item`, `Rename-Item` and `Copy-Item` cmdlets. This change does not affect the `New-Item` cmdlet, as it is able to create files with names containing Unix globbing characters.

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 [hunk 1]
```diff
@@ -246,6 +246,49 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
...
+Describe "Handling of globbing patterns" -Tags "CI" {
+
+    Context "Handling of Unix [ab] globbing patterns in literal paths" {
+        BeforeAll {
+            $filePath = Join-Path $TESTDRIVE "file[txt].txt"
+            $newPath = Join-Path $TESTDRIVE "file.txt.txt"
+            $dirPath = Join-Path $TESTDRIVE "subdir"
+        }
+        BeforeEach {
+            $file = New-Item -ItemType File $filePath -Force
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: Please add `-Path` in all places wherever flowed.

**Review by iSazonov — COMMENTED** [review]
@jeffbi Original Issue say `*-Item` but the PR fix only Rename-Item. Please add comment about this in the PR description.

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 (hunk 1 above)
Comment (jeffbi) [inline test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: Fixed.

**Comment by iSazonov** [comment]
@jeffbi 
>This affects the Move-Item, Remove-Item, Rename-Item and Copy-Item cmdlets.

What about `New-Item`?

**Comment by jeffbi** [comment]
@iSazonov The issue didn't affect `New-Item`. `New-Item` was able to create the file `file[txt].txt`, it was the other cmdlets that couldn't find the file once it was created.

**Comment by iSazonov** [comment]
@jeffbi My question was again about addition to the PR description. I expect that this will help make PowerShell documentation more precise. Thanks!

**Comment by jeffbi** [comment]
@iSazonov OK, I've updated the PR description to include a note about `New-Item`

**Comment by iSazonov** [comment]
@jeffbi Many thanks!

LGTM.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Pull request merged**